### DISCUSSION TITLE
How do I inspect the current active OTel settings?

### DISCUSSION THREAD

**Discussion by RoadRunnr** [open-telemetry/opentelemetry-erlang#536] [Q&A]
How do I troubleshoot the OTel settings from within a running Erlang VM?
Are there API calls (I haven't found any) to inspect the configuration (which processor is unse, what are it settings, which exporter is in use, what are its settings), running process and the state of connections in the exporter?

**Comment by tsloughter** [comment]
@RoadRunnr there isn't an API function for this but you could get the state of the global tracer provider, which stores the processors with:

`sys:get_state(otel_tracer_provider_global).`

But I just checked to be sure and sadly the processors are only stored as anonymous functions so won't be much help there.

I'd be interested in improving this for sure.
